test/specs/helpers.spec.js

```diff
@@ -441,5 +443,34 @@ define([
                 });
             });
         });
+
+        describe('locale', () => {
+            it('should return a rejected promise upon an invalid locale', (done) => {
+                expectAsync(locale.setDefaultLocale('foo-bar')).toBeRejectedWith('Please pass in a valid locale string (az-AZ) or locale object definition');
+                done();
+            });
+
+            it('should return a resolved promise when a valid locale definition is given', (done) => {
+                const validLocaleDefinition = {
+                    'decimal': '.',
+                    'thousands': ',',
+                    'grouping': [
+                      3
+                    ],
+                    'currency': [
+                      '$',
+                      ''
+                    ]
+                  };
+                expectAsync(locale.setDefaultLocale(validLocaleDefinition)).toBeResolved();
+                done();
+            });
+
+            it('should return a rejected promise when a invalid locale definition is given', (done) => {
+                const invalidLocaleDefinition = {};
+                expectAsync(locale.setDefaultLocale(invalidLocaleDefinition)).toBeRejected();
+                done();
+            });
+        });
```

Comment:
I would add a couple of test cases for the two OK cases.

Reply:
I had those tests, but since i wasn't able to mock the d3.json call, i removed those tests since i would be relying on an internet connection to make the test pass. If that's not a problem, i could re add the OK cases test.

Reply:
Makes sense, add those here commented for now, I will take a look later.

Reply:
Ok, re-added them